//
// Copyright 2013-2019 Will Mason
// 
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
// 
//      http://www.apache.org/licenses/LICENSE-2.0
// 
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
//

package chucho.proto;

message log_events {
    repeated log_event events = 1;
    optional string host_name = 2;
}

message log_event {
    optional string formatted_message = 1;
    optional uint64 seconds_since_epoch = 2;
    optional string file_name = 3;
    optional uint32 line_number = 4;
    optional string function_name = 5;
    optional string logger = 6;
    optional string level_name = 7;
    optional string marker = 8;
    optional string thread = 9;
}
